src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java

```diff
@@ -237,4 +241,63 @@ static IngestedFileStatus convertIngestStatus(IngestStatus status)
     LOAD_IN_PROGRESS,
     NOT_FOUND,
   }
+
+  /**
+   * Interfaces to define the lambda function to be used by backoffAndRetry
+   */
+  interface backoffFunction
+  {
+    Object operate() throws Exception;
+  }
+  interface backoffFunctionWithParameter
+  {
+    Object operate(Object obj) throws Exception;
+  }
+
+  /**
+   * Backoff and retry logic
+   */
+  public static Object backoffAndRetry(SnowflakeTelemetryService telemetry, backoffFunction runnable) throws Exception
+  {
+    int times = 0;
+    while (true)
+    {
+      try
+      {
+        return runnable.operate();
+      }
+      catch (Exception e)
+      {
+        telemetry.reportSnowflakeThrottle(e.getMessage());
+        if (times >= backoffSec.length)
+        {
+          throw e;
+        }
+      }
+      Thread.sleep(backoffSec[times] * 1000);
+      times = times + 1;
+    }
+  }
+  public static Object backoffAndRetry(SnowflakeTelemetryService telemetry, backoffFunctionWithParameter runnable,
+                                       Object input) throws Exception
+  {
+    int times = 0;
```

Comment:
We can combine these two functions, a lot duplicate code.

Reply:
ack